client/state/timezones/reducer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	DESERIALIZE,
+	SERIALIZE,
+
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS,
+} from 'state/action-types';
+import {
+	rawOffsetsSchema,
+	timezonesSchema
+} from './schema';
+
+const offsets = createReducer( [], {
+	[ TIMEZONES_RECEIVE ]: ( state, { rawOffsets = [] } ) => rawOffsets.slice( 0 )
+}, rawOffsetsSchema );
+
+const timezones = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, { timezonesByContinent } ) => ( timezonesByContinent )
+}, timezonesSchema );
+
+export const items = combineReducers( {
```

Comment:
Is any of this data redundant? If so I would recommend storing the flat timezones, and created [memoized selectors](https://github.com/Automattic/wp-calypso/tree/master/client/lib/create-selector) for easy lookup by continent